tasks/eth/013-fp-upgrade-fjord/NestedSignFromJson.s.sol

```diff
@@ -0,0 +1,85 @@
+
+contract NestedSignFromJson is OriginalNestedSignFromJson {
+    using LibString for string;
+
+    DisputeGameFactory dgfProxy;
+
+    // Safe contract for this task.
+    address immutable proxyAdminOwnerSafe = vm.envAddress("OWNER_SAFE");
+    address immutable securityCouncilSafe = vm.envAddress("COUNCIL_SAFE");
+    address immutable foundationUpgradesSafe = vm.envAddress("FOUNDATION_SAFE");
+
+    // New dispute game implementations
+    // See governance proposal for verification - https://gov.optimism.io/t/upgrade-proposal-9-fjord-network-upgrade/8236
+    FaultDisputeGame constant faultDisputeGame = FaultDisputeGame(0xf691F8A6d908B58C534B624cF16495b491E633BA);
+    PermissionedDisputeGame constant permissionedDisputeGame = PermissionedDisputeGame(0xc307e93a7C530a184c98EaDe4545a412b857b62f);
+
+    /// @notice Sets up the dgfProxy 
+    function setUp() public {
+        string memory addressesJson;
+        // Read addresses json
+        try vm.readFile(
+            string.concat(vm.projectRoot(), "/lib/superchain-registry/superchain/extra/addresses/mainnet/op.json")
+        ) returns (string memory data) {
+            addressesJson = data;
+        } catch {
+            revert("Failed to read lib/superchain-registry/superchain/extra/addresses/mainnet/op.json");
+        }
+        dgfProxy = DisputeGameFactory(stdJson.readAddress(addressesJson, "$.DisputeGameFactoryProxy"));
+    }
+
+    function getCodeExceptions() internal pure override returns (address[] memory) {
+        // No code exceptions expected
+    }
+
+    function _postCheck(Vm.AccountAccess[] memory accesses, SimulationPayload memory) internal view override {
+        console.log("Running post-deploy assertions");
+        checkStateDiff(accesses);
+        checkDGFProxy();
+        console.log("All assertions passed!");
+    }
+
+    function checkStateDiff(Vm.AccountAccess[] memory accountAccesses) internal view override {
+        require(accountAccesses.length > 0, "No account accesses");
+        super.checkStateDiff(accountAccesses);
+
+        for (uint256 i; i < accountAccesses.length; i++) {
+            Vm.AccountAccess memory accountAccess = accountAccesses[i];
+
+            // Assert that only the expected accounts have been written to.
+            for (uint256 j; j < accountAccess.storageAccesses.length; j++) {
+                Vm.StorageAccess memory storageAccess = accountAccess.storageAccesses[j];
+                if (storageAccess.isWrite) {
+                    address account = storageAccess.account;
+                    require(
+                        // DisputeGameFactoryProxy is expected to be updated
+                        account == address(dgfProxy)
+                        // State changes the Safe's are also expected.
+                        || account == proxyAdminOwnerSafe || account == securityCouncilSafe
+                            || account == foundationUpgradesSafe,
+                        "state-100"
```

Comment:
I get a revert with `state-100` when simulating. This is because we also have state changes in the LivenessGuard anytime the SC signs a transaction, which you can see in the tenderly simulation

Reply:
As clarified in https://github.com/ethereum-optimism/superchain-ops/pull/232#discussion_r1649226929, this will only show when simulating as the council, not as the foundation